serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -78,10 +82,21 @@ def _is_safe_to_sample(prob_like):
     logits_greedy = logits[mask_greedy]
 
     if logits_greedy.shape[0] > 0:
-        res_greedy = torch.argmax(logits_greedy, -1).cpu().numpy()
-
+        # Greedy sampling
+        logprobs = torch.log_softmax(logits_greedy, dim=-1)
```

Comment:
It is better to introduce a new sample function for the logprob case. When logprob is not requested, there should be no perf regression.